Workade/Managers/FirebaseManager.swift

```diff
@@ -128,9 +128,9 @@ final class FirebaseManager: NSObject {
                 
                 if let user = result?.user {
                     Task { [weak self] in
-                        if var userInfo = try await FirestoreDAO.shared.getUser(userID: user.uid) {
-                            if let region = self?.region {
-                                try await FirestoreDAO.shared.createActiveUser(user: ActiveUser(id: userInfo.id, job: userInfo.job, region: region, startDate: .now))
+                        if UserManager.shared.user.value != nil {
+                            if let region = self?.region, let userInfo = UserManager.shared.user.value {
+                                try await FirestoreDAO.shared.createActiveUser(user: ActiveUser(id: userInfo	.id, job: userInfo.job, region: region, startDate: .now))
                                 try await FirestoreDAO.shared.updateUser(user: User(id: userInfo.id, name: userInfo.name, email: userInfo.email, job: userInfo.job, activeRegion: region))
                             }
                             DispatchQueue.main.async {
```

Comment:
[P2] 이 부분도 위와 마찬가지로 if let 구문을 한 줄로 줄일 수 있을 것 같아요~! 그리고 signupCompletion() 클로저 호출이 if든 else든 호출되는 것 같아서 한 번만 써줘도 될 것 같아용~!
```suggestion
Task { [weak self] in
                        if let userInfo = UserManager.shared.user.value, let region = self?.region {
                            try await FirestoreDAO.shared.createActiveUser(user: ActiveUser(id: userInfo    .id, job: userInfo.job, region: region, startDate: .now))
                            try await FirestoreDAO.shared.updateUser(user: User(id: userInfo.id, name: userInfo.name, email: userInfo.email, job: userInfo.job, activeRegion: region))
                        }
                        DispatchQueue.main.async {
                            signinCompletion()
                        }
                    }
```

Reply:
위에 if let은 줄였지만 singupCompletion과 SigninCompletion은 서로 달라서 어쩔 수 없이 if else에 따로 선언해야 할 것 같습니다.

Reply:
아아 up과 in이였군요 😂 잘못봤네용 🥹 ㅜㅜ